### PR TITLE
Perf: optimize rhog_symmetry

### PR DESCRIPTION
**Pull request by maki49**
solveing issue  #2168

### CODE REVIEW

File: source/module_cell/module_symmetry/symmetry.cpp [hunk 1]
```diff
@@ -1405,21 +1398,23 @@ void Symmetry::rhog_symmetry(std::complex<double> *rhogtot,
...
-                        tmp_gdirect_double.y=(double)tmp_gdirect.y;
-                        tmp_gdirect_double.z=(double)tmp_gdirect.z;
+                        tmp_gdirect_double = tmp_gdirect_double * ModuleBase::TWO_PI;
                         double cos_arg=0.0, sin_arg=0.0;
                         // for each pricell in supercell:
                         for(int ipt=0;ipt<this->ncell;++ipt)
                         {
-                            arg = ( tmp_gdirect_double * (gtrans[invmap[isym]]+ptrans[ipt]) ) * ModuleBase::TWO_PI;
+                            arg = tmp_gdirect_double * (gtrans[invmap[isym]]+ptrans[ipt]) ;
                             cos_arg += cos(arg);
```
Comment (dyzheng) [inline source/module_cell/module_symmetry/symmetry.cpp]: I guess the most expensive code is sin() and cos(), have you tested the improvement of this PR?

Comment (dyzheng) [reply source/module_cell/module_symmetry/symmetry.cpp]: #1780 can help for this.

**Review by dyzheng — APPROVED** [review]

**Comment by maki49** [comment]
I extracted the gtrans-phase-factor outside of the ptrans-loop which make it slightly faster. 

I find that the most time-consuming is these two adds in the ptrans-loop (that's what I left out in the last commit). 
But now I have no idea to optimize this. 
image
Or maybe I could add an `if()` before the the ptrans-loop to skip it when there are too many primitive cells? It just make the symmetrization incomplete and hardly affect the results.

**Comment by dyzheng** [comment]
I will merge this PR and test the problem you mentioned.

**Pull request merged**